### PR TITLE
Fix errors due to misplaced `read_data.py` and NionSwift 16.2

### PR DESCRIPTION
**Pull request by francisco-dlp**
In my installation, using Nion Swift 16.2, I need the fixes in this PR to run the plugin.

In particular:

* The path of the `read_data.py` file was wrong
* In NS 16.2, ``frame_parameters`` is not a ``dict``, so things like ``if "acquisition_mode" in frame_parameters:`` don't work.

### CODE REVIEW

**Review by yvesauad — COMMENTED** [review]
Hi Francisco!
Thanks for these. So actually this code is running in VG Cold, Lumiere and partially in ChromaTEM. I agree that installing using pip is much better, but currently i dont know how to handle the submodules properly. The easiest way i found is by using multiple setup files. The issue with read data is that this is a new stuff I've implemented and I havent added this .py in your lille setup. I will stay with ..aux_files.config because i am manually putting this file inside config using the setup. 

Regarding the dictionary, this is a pretty good solution! in fact this is not a issue with nionswift 0.16.2 but actually with nion instrumentation 0.20.4. As you are already using the most up-to-date software I will follow you. Thanks! 

In short: i will reject this PR because i will fix your setupLille.py file. I agree this is a bad solution from myself but i can explain it better why it must be like this from now (basically i can use pip install but rather python install setupXXX.py)

**Comment by francisco-dlp** [comment]
I see, I didn't realize that you were installing a `read_data.py` file as data file. Wouldn't it be better to place it somewhere else among the Python files?

By the way, would it be possible to separate the code in this repository into multiple packages, one per device? In this way, instead of hacking ``setup.py`` you could install in each machine the packages that you need.

Also, would it be possible to maintain just one NS scan code? (We have Marcel's one installed in our machine, but that one doesn't recognize the TIMEPIX.)

Of course we're are happy to help with these tasks.

**Comment by francisco-dlp** [comment]
Sorry, I didn't realize that I have also added the ``setup.py`` file that I used to install the package with ``pip``.

**Comment by yvesauad** [comment]
Yes to all of them hahaha

i) I agree read_data must be better inside one of those files. However 
there is no single package that go to all microscopes. The only files 
that go to all machines are the config files so thats why i have put it 
there. But maybe it is actually better to have a dedicated package to 
read_data now.

ii) I actually dont know how to implement this properly. If I do one 
package per machine, then I need to maintain twice the same file, no? I 
am pretty sure i am missing something because it shouldn't be that 
difficult. Is there a way to keep a single scan file for multiple 
packages, for example?

iii) This is a current project with Marcel. Should be very doable but we 
didn't take the time to do it!

PS: I am answering you now regarding the simulated data.

cheers,

Yves

Le 5/20/2022 à 3:09 PM, Francisco de la Peña a écrit :
